roll/__init__.py

```diff
@@ -122,19 +149,38 @@ def add_route(view):
                     raise ValueError(f"Empty view: {view}")
             else:
                 if methods is None:
-                    methods = ['GET']
+                    methods = ["GET"]
                 payload = {method: view for method in methods}
             payload.update(extras)
             if protocol_class.ALLOWED_METHODS:
                 assert set(methods) <= set(protocol_class.ALLOWED_METHODS)
             self.routes.add(path, **payload)
+            self._register_route_name(path, view, extras.get("name"))
             return view
 
         return add_route
 
+    def _register_route_name(self, path: str, view, name: str = None):
+        cleaned = CLEAN_PATH_PATTERN.sub("", path)
+        if not name:
+            name = view.__name__.lower()
+        if name in self._urls:
+            _, view = self._urls[name]
+            ref = f"{view.__module__}.{view.__name__}"
+            raise DuplicateRouteName(f"Route with name {name} already exists: {ref}")
```

Comment:
I see that elsewhere in the code we only use base exception (ValueError, AttributeError), so I think we are on consistent with those custom exceptions.

I'm tempted for now to only use base exception everywhere, including here (ValueError), but we can also have named exception in the other places (but that's a bit out of this PR IMHO).

Thoughts ?

Reply:
Let's stick to base exceptions for now, and add an issue related to this?

Reply:
Done and created #137